### PR TITLE
#357 - Upgrade locus-api-android to version 0.9.50. Fixed Locus integration.

### PR DESCRIPTION
**Pull request by Dratenik**
locus-api-android dependency has been updated to latest version.

## Description

In connection with this, it was necessary to modify many occurrences of classes that changed their name and location.

## Related issues
fixes https://github.com/cgeo/WhereYouGo/issues/357

## Additional context
[locus-api - CHANGELOG](https://github.com/asamm/locus-api/blob/master/CHANGELOG.md): it's not complete, but it's a good starting point.

### CODE REVIEW

File: src/main/java/menion/android/whereyougo/gui/activity/CartridgeDetailsActivity.java [hunk 1]
```diff
@@ -74,9 +74,7 @@ public void onCreate(Bundle savedInstanceState) {
 
         TextView tvDistance = (TextView) findViewById(R.id.layoutDetailsTextViewDistance);
 
-        Location loc = new Location(TAG);
-        loc.setLatitude(MainActivity.cartridgeFile.latitude);
-        loc.setLongitude(MainActivity.cartridgeFile.longitude);
+        Location loc = new Location(MainActivity.cartridgeFile.latitude, MainActivity.cartridgeFile.longitude);
```
Comment (bekuno) [inline src/main/java/menion/android/whereyougo/gui/activity/CartridgeDetailsActivity.java:77]: The 'TAG' string is now unused. Please remove.

File: src/main/java/menion/android/whereyougo/gui/extension/DataInfo.java [hunk 2]
```diff
@@ -175,9 +175,7 @@ public DataInfo setImageRight(Bitmap image) {
     }
 
     public Location getLocation() {
-        Location loc = new Location(TAG);
-        loc.setLatitude(value01);
-        loc.setLongitude(value02);
+        Location loc = new Location(value01, value02);
```
Comment (bekuno) [inline src/main/java/menion/android/whereyougo/gui/extension/DataInfo.java:178]: The 'TAG' string is now unused. Please remove.

File: src/main/java/menion/android/whereyougo/maps/utils/LocusMapDataProvider.java [hunk 3]
```diff
@@ -28,23 +35,16 @@
...
 import cz.matejcik.openwig.Zone;
 import cz.matejcik.openwig.formats.CartridgeFile;
-import locus.api.android.objects.PackWaypoints;
-import locus.api.objects.extra.ExtraData;
-import locus.api.objects.extra.ExtraStyle;
-import locus.api.objects.extra.ExtraStyle.LineStyle.ColorStyle;
-import locus.api.objects.extra.ExtraStyle.LineStyle.Units;
 import locus.api.objects.extra.Location;
-import locus.api.objects.extra.Track;
-import locus.api.objects.extra.Waypoint;
```
Comment (bekuno) [inline src/main/java/menion/android/whereyougo/maps/utils/LocusMapDataProvider.java:39]: Please reorder the imports.
Also in the other files, I would say.

File: src/main/java/menion/android/whereyougo/maps/utils/LocusMapDataProvider.java [hunk 4]
```diff
@@ -75,13 +75,12 @@ public void addCartridges(Vector<CartridgeFile> cartridges) {
...
-            loc.setLongitude(cartridge.longitude);
-            Waypoint wpt = new Waypoint(cartridge.name, loc);
-            wpt.addParameter(ExtraData.PAR_DESCRIPTION, cartridge.description);
-            wpt.addUrl(cartridge.url);
-            pack.addWaypoint(wpt);
+            Location loc = new Location(cartridge.latitude,cartridge.longitude);
+            Point wpt = new Point(cartridge.name, loc);
+            wpt.addParameter(GeoDataExtra.PAR_DESCRIPTION, cartridge.description);
+            wpt.addParameter(GeoDataExtra.PAR_SOURCE, cartridge.url);
+            //TODO wpt.addUrl(cartridge.url);
```
Comment (bekuno) [inline src/main/java/menion/android/whereyougo/maps/utils/LocusMapDataProvider.java:82]: The ``addUrl` was first migrated to `addParameterUrl`.
In version 0.9.50 this was moved to `GeoDataHelper`, 
see https://github.com/asamm/locus-api/commit/ecfd577b950780162d5bdd7f1505759618637645
So I asume that `GeoDataExtra.PAR_SOURCE` isn't correct.

File: src/main/java/menion/android/whereyougo/maps/utils/LocusMapDataProvider.java [hunk 5]
```diff
@@ -99,18 +98,16 @@ public void addZone(Zone z, boolean mark) {
...
         if (locs.size() >= 3)
             locs.add(locs.get(0));
 
         Track track = new Track();
-        ExtraStyle style = new ExtraStyle();
-        style.setLineStyle(ColorStyle.SIMPLE, Color.MAGENTA, 2.0f, Units.PIXELS);
-        track.styleNormal = style;
+        GeoDataStyle style = new GeoDataStyle();
+        style.setLineStyle(Color.MAGENTA, 2.0f);
+        track.setStyleNormal(style);
```
Comment (bekuno) [inline src/main/java/menion/android/whereyougo/maps/utils/LocusMapDataProvider.java:110]: I'm not sure, if this is enough.
My solution would be
~~~
        LineStyle lineStyle = new LineStyle();
        lineStyle.setColoring(LineStyle.Coloring.SIMPLE);
        lineStyle.setColorBase(Color.MAGENTA);
        lineStyle.setWidth(2.0f);
        lineStyle.setUnits(LineStyle.Units.PIXELS);

        GeoDataStyle geoDataStyle = new GeoDataStyle();
        geoDataStyle.setLineStyle(lineStyle);
~~~

**Review by bekuno — CHANGES_REQUESTED** [review]

File: src/main/java/menion/android/whereyougo/gui/activity/CartridgeDetailsActivity.java (hunk 1 above)
Comment (Dratenik) [reply src/main/java/menion/android/whereyougo/gui/activity/CartridgeDetailsActivity.java:75]: Done in 2daea5a655714047859cba8bb48e8b816def4518

File: src/main/java/menion/android/whereyougo/gui/extension/DataInfo.java (hunk 2 above)
Comment (Dratenik) [reply src/main/java/menion/android/whereyougo/gui/extension/DataInfo.java:177]: Done in 2daea5a655714047859cba8bb48e8b816def4518

File: src/main/java/menion/android/whereyougo/maps/utils/LocusMapDataProvider.java (hunk 3 above)
Comment (Dratenik) [reply src/main/java/menion/android/whereyougo/maps/utils/LocusMapDataProvider.java:42]: Done in 2daea5a655714047859cba8bb48e8b816def4518. Surprisingly, Idea can't do this by default. You need to install and activate the plugin.

File: src/main/java/menion/android/whereyougo/maps/utils/LocusMapDataProvider.java (hunk 4 above)
Comment (Dratenik) [reply src/main/java/menion/android/whereyougo/maps/utils/LocusMapDataProvider.java]: Used `GeoDataHelper` not working for me. In 0a12e7b151d35b4c236698e4288a36214d6d8c21 use `Point.setExtraData` directly.

File: src/main/java/menion/android/whereyougo/maps/utils/LocusMapDataProvider.java (hunk 5 above)
Comment (Dratenik) [reply src/main/java/menion/android/whereyougo/maps/utils/LocusMapDataProvider.java]: Thanks for this. Used in 0a12e7b151d35b4c236698e4288a36214d6d8c21 and works.

File: src/main/java/menion/android/whereyougo/geo/location/Location.java [hunk 6]
```diff
@@ -7,10 +7,13 @@ public Location() {
...
     public Location(android.location.Location loc) {
-        this(loc.getProvider(), loc.getLatitude(), loc.getLongitude());
+        this(loc.getLatitude(), loc.getLongitude());
         setTime(loc.getTime());
         if (loc.hasAccuracy()) {
-            setAccuracy(loc.getAccuracy());
+            setAccuracyHor(loc.getAccuracy());
+        }
+        if(loc.hasVerticalAccuracy()) {
+            setAccuracyVer(loc.getAccuracy());
```
Comment (bekuno) [inline src/main/java/menion/android/whereyougo/geo/location/Location.java:16]: should be `            setAccuracyVer(loc.getVerticalAccuracyMeters());` ?

Maybe the implementation of `AccuracyVer` would be better in another PR.
There are some places where could be used, e.g. in `LocationState`.

**Review by bekuno — CHANGES_REQUESTED** [review]

File: src/main/java/menion/android/whereyougo/geo/location/Location.java (hunk 6 above)
Comment (Dratenik) [reply src/main/java/menion/android/whereyougo/geo/location/Location.java]: I agree, taken in 204f43dae38ae08ac69201ff99cea7d96cf5cfac.

**Review by bekuno — APPROVED** [review]

**Comment by bekuno** [comment]
@Dratenik Many thanks for your contribution.
